TypeCobol.LanguageServer/WorkspaceProjectStore.cs

```diff
@@ -0,0 +1,286 @@
+            System.Diagnostics.Debug.Assert(fromWorkspaceProject.Contains(docContext));
+            System.Diagnostics.Debug.Assert(!toWorkspaceProject.Contains(docContext));
+
+            fromWorkspaceProject.RemoveDocument(docContext);
+            toWorkspaceProject.AddDocument(docContext);            
+        }
+
+        /// <summary>
+        /// Associate a document to a Workspace Project
+        /// </summary>
+        /// <param name="docUri">The Uri of the document to be associated to workspace projet</param>
+        /// <param name="projectKey">The project key to which the document shall be associated</param>
+        /// <param name="copyFolders">List of Copy Folders to be associated with the underlying CompilationProject instance.</param>
+        /// <param name="workspace">The Workspace project instance</param>
+        internal void AssociateDocumentToProject(Uri docUri, string projectKey, List<string> copyFolders, Workspace workspace)
+        {
+            // Check if the docUri is already associated to a WorkspaceProject
+            DocumentContext docContext = null;
+            WorkspaceProject workspaceProject = null;
+            if (!TryGetOpenedWorkspaceDocumentProjet(docUri, out docContext, out workspaceProject))
+            {   // hum... It is a new file without corresponding DidOpenDocument notification sent
+                // Ignore such file.
+                return;
+            }
+            // Now try to match its new WorkspaceProject project it can be the same
+            WorkspaceProject newWorkspaceProject = null;
+            if (projectKey == null)
+            {   // No project key specified ==> Goes in to the Default Workspace Project.
+                newWorkspaceProject = DefaultWorkspaceProject;
+            }
+            else
+            {   // Find the target WorkspaceProject
+                newWorkspaceProject = FindProject(projectKey);
+                if (newWorkspaceProject == null)
+                { // The target project does not exists ==> We must create this new workspace project.
+                    newWorkspaceProject = CreateWorkspaceProject(projectKey, workspace.Configuration);
+                }
+            }
+            if (workspaceProject != newWorkspaceProject)
+            {
+                // We must remove this document from its current workspace project and add it into its new workspace project.
+                MoveWorkspaceProjectDocument(docContext, workspaceProject, newWorkspaceProject);
+            }
+            bool refreshAll = newWorkspaceProject.UpdateSourceFileProvider(copyFolders, workspace.Configuration);
+            if ((workspaceProject != newWorkspaceProject) && !refreshAll)
+            {   // Only this Document must be refreshed
+                workspace.RefreshOpenedDocument(docContext, true);
+            }
+            else if (refreshAll)
+            {   // All documents of the WokspaceProject must be refreshed.
```

Comment:
typo: 'WorkspaceProject'

Reply:
Changed in [e1c296c](https://github.com/TypeCobolTeam/TypeCobol/pull/2145/commits/e1c296c5835196790db878e8efdab69f6a196c18)